docs/proposals/crt_management_proposal.md

```diff
@@ -0,0 +1,113 @@
+
+container_runtime logic is currently spread across 3 roles: docker, openshift_docker,
+and openshift_docker_facts.  The name 'docker' does not accurately portray what
+the role(s) do.
+
+## Rationale
+* Refactor docker (and related meta/fact roles) into 'container_runtime' role.
+* Strip all meta-depends on container runtime out of other roles and plays.
+* Create a 'prerequisites.yml' entry point that will setup various items
+such as container storage and container runtime before executing installation.
+* All other roles and plays should merely consume container runtime, should not
+configure, restart, or change the container runtime as much as feasible.
+
+## Design
+
+The container_runtime role should be comprised of 3 'pseudo-roles' which will be
+consumed using include_role; each component area should be enabled/disabled with
+a boolean value, defaulting to true.
+
+I call them 'pseudo-roles' because they are more or less independent functional
+areas that may share some variables and act on closely related components.  This
+is an effort to reuse as much code as possible, limit role-bloat (we already have
+an abundance of roles), and make things as modular as possible.
+
+```yaml
+# prerequisites.yml
+- include: std_include.yml
+- include: container_runtime_setup.yml
+...
+# container_runtime_setup.yml
+- hosts: "{{ openshift_runtime_manage_hosts | default('oo_nodes_to_config') }}"
+  tasks:
+    - include_role:
+        name: container_runtime
+        tasks_from: install.yml
+      when: openshift_container_runtime_install | default(True) | bool
+    - include_role:
+        name: container_runtime
+        tasks_from: storage.yml
+      when: openshift_container_runtime_storage | default(True) | bool
+    - include_role:
+        name: container_runtime
+        tasks_from: configure.yml
+      when: openshift_container_runtime_configure | default(True) | bool
+```
+
+Note the host group on the above play.  No more guessing what hosts to run this
+stuff against.  If you want to use an atomic install, specify what hosts will need
+us to setup container runtime (such as etcd hosts, loadbalancers, etc);
+
+We should direct users that are using atomic hosts to disable install in the docs,
+let's not add a bunch of logic.
+
+Alternatively, we can create a new group.
+
+### Part 1, container runtime install
+Install the container runtime components of the desired type.
+
+```yaml
+# install.yml
+- include: docker.yml
+  when: openshift_container_runtime_install_docker | bool
+
+- include: crio.yml
+  when: openshift_container_runtime_install_crio | bool
+
+... other container run times...
+```
+
+Alternatively to using booleans for each run time, we could use a variable like
+"openshift_container_runtime_type".  This would be my preference, as we could
+use this information in later roles.
+
+### Part 2, configure/setup container runtime storage
+Configure a supported storage solution for containers.
```

Comment:
We as an organization need to determine our path forward with the recommended storage solution.